src/stats/hist.jl

```diff
@@ -79,7 +79,10 @@ function Makie.plot!(plot::StepHist)
     pop!(attr, :scale_to)
     pop!(attr, :bins)
     # plot the values, not the observables, to be in control of updating
-    stairs!(plot, points[]; attr..., color=color)
+    st = stairs!(plot, points[]; attr..., color=color)
+    onany(plot.normalization, plot.scale_to, plot.weights, plot.bins) do _, _, _, _
+        st[1][] = points[]
+    end
```

Comment:
```suggestion
    stairs!(plot, points; attr..., color=color)
    plot
```
I got the chance to look at this again and think about it some more and the more I consider it the more I realize that we should probably just pass the `points` Observable here. The reason this isn't done for `hist` seems to be because `widths` depends upon some of the same Observables as `points`, so the trick to avoid redundant updates is employed there.

However, `stephist` is different, there's no `widths` so there doesn't appear to be any efficiency gains by managing updates separately. Any updates to `bins`, `normalization`, `scale_to`, and `weights` will update `points` so the plot updating is handled correctly.